src/utils/selectors.ts

```diff
@@ -1,4 +1,5 @@
 import { FirebaseReducer } from "react-redux-firebase";
+import { mapValues } from "lodash";
```

Comment:
[OOS] I suggest removing `lodash` as dependency and adding only the required functions as modules. The library is 1.4mb large according to the npmjs https://www.npmjs.com/package/lodash

Adding only the functions that we need will be much less IMO, we need 4 or 5 functions only.
https://www.npmjs.com/package/lodash.chunk
https://www.npmjs.com/package/lodash.isequal
https://www.npmjs.com/package/lodash.throttle
etc...

Reply:
Importing the individual npm packages like that is a legacy method that's only really useful for libraries/environments that aren't using bundling/tree shaking/etc.

---

@goran-peoski-work Put up https://github.com/sparkletown/sparkle/pull/1715 for replacing `lodash` with `lodash-es`, which is a more modern/compatible way of solving this (and with less manual effort required) when using bundlers/tree shaking/etc.

---

I've also been looking into a few further optimisations as well lately, since even if we don't include `lodash` ourselves, a good chunk of our dependencies do, so it will end up in our bundle regardless. A few notes/resources off the top of my head:

- https://github.com/gsoft-inc/craco
  - Needed for customising the babel/webpack/etc config's with ejecting from `create-react-app`
- https://github.com/lodash/babel-plugin-lodash
  - > Modular Lodash builds without the hassle.
  - > A simple transform to cherry-pick Lodash modules so you don’t have to. Combine with lodash-webpack-plugin for even smaller cherry-picked builds!
- https://github.com/lodash/lodash-webpack-plugin
  - > Create smaller Lodash builds by replacing feature sets of modules with noop, identity, or simpler alternatives. This plugin complements `babel-plugin-lodash` by shrinking its cherry-picked builds even further!
  - > DISCLAIMER: Using this plugin without enabling the proper feature sets may cause lodash functions to behave in unexpected ways. Methods may appear to work, however they might return incorrect results.
- https://github.com/GoogleChromeLabs/webpack-libs-optimizations
  - > Using a library in your webpack project? Use these tips to make your bundle smaller!
  - https://github.com/GoogleChromeLabs/webpack-libs-optimizations/issues/21#issuecomment-756622581
    - Concludes that with webpack 4, using `lodash` is actually better than `lodash-es` when combined with `babel-plugin-lodash` & `lodash-webpack-plugin`
  - https://github.com/GoogleChromeLabs/webpack-libs-optimizations/issues/25
    - Has further discussions/nuances on why that might not be the best idea/potential alternatives.
    - https://github.com/GoogleChromeLabs/webpack-libs-optimizations/issues/25#issuecomment-874370372
      - My thoughts/comments, including pointing out that `create-react-app` currently still uses webpack v4, which means the points made in https://github.com/GoogleChromeLabs/webpack-libs-optimizations/issues/21#issuecomment-756622581 about tree shaking performance likely still apply to us

Reply:
Also cross-posted the above in https://github.com/sparkletown/sparkle/pull/1715#issuecomment-874371412 for reference

Reply:
This is implemented in https://github.com/sparkletown/sparkle/pull/1735